tests/fsharp/Compiler/Conformance/BasicGrammarElements/BasicConstrants.fs

```diff
@@ -0,0 +1,66 @@
+﻿// Copyright (c) Microsoft Corporation.  All Rights Reserved.  See License.txt in the project root for license information.
+
+namespace FSharp.Compiler.UnitTests
+
+open NUnit.Framework
+open FSharp.Compiler.SourceCodeServices
+
+[<TestFixture>]
+module ``Basic Grammar Element Constraints`` =
+
+    [<Test>]
+    let `` Basic constraints compile `` () =
+        CompilerAssert.Pass 
+            """
+let sbyteConst = 1y
+let int16Const = 1us
+let int32Const = 1ul
+let int64Const = 1UL
+    
+let byteConst  = 1uy
+let uint16Const = 1us
+let uint32Const = 1ul
+let uint64Const = 1uL
+    
+let ieee32Const1 = 1.0f
+let ieee32Const2 = 1.0F
+let ieee32Const3 = 0x0000000000000001lf
+    
+let ieee64Const1 = 1.0
+let ieee64Const2 = 0x0000000000000001LF
+    
+let bigintConst = 1I
+    
+// let bignumConst = 1N - you need a reference to PowerPack.dll now
+    
+let decimalConst1 = 1.0M
+let decimalConst2 = 1.0m
+    
+let charConst = '1'
+    
+let stringConst = "1"
+    
+let bytestringConst = "1"B
+    
+let bytecharConst = '1'B
+    
+let boolConst1 = true
+let boolConst2 = false
+    
+let unitConst = ()
+            """
+
+    [<Test>]
+    let ``int 64``() = 
+        let creditCardNumber = 1234_5678_9012_3456L
```

Comment:
I think this needs to be given to one of the CompilerAssert.Parse … functions.

The purpose of the test is to ensure that the compiler compiles it correctly, however, the unit tests are built with proto compiler, not the product compiler.

Reply:
@KevinRansom  thanks for the feedback I checked a few other PR's and I think I know what to do now I will change the code to use the correct compiler.
I think its CompilerAssert.CompileExeAndRun  and have an  if <>  exit 1 in the script.

Reply:
@KevinRansom @cartermp   I updated the PR and changed the name to reflect a smaller scope as you suggested. The PR now only includes 1 file change. 
would you suggest I have the follow-up PR's include these changes or have each of them branch from master?